lib/xcode.js

```diff
@@ -38,9 +38,22 @@ async function getPathFromSymlink (failMessage) {
   // xcode-select allows users to override its settings with the DEVELOPER_DIR env var,
   // so check that first
   if (util.hasContent(env.DEVELOPER_DIR)) {
-    return hasExpectedSubDir(env.DEVELOPER_DIR) ? env.DEVELOPER_DIR : env.DEVELOPER_DIR + XCODE_SUBDIR;
+    const customPath = hasExpectedSubDir(env.DEVELOPER_DIR) ?
+                                         env.DEVELOPER_DIR  :
+                                         env.DEVELOPER_DIR + XCODE_SUBDIR;
+
+    if (await fileExists(customPath)) {
+      return customPath;
+    } else {
+      let mesg = `Could not find path to Xcode, environment variable ` +
+                 `DEVELOPER_DIR set to: ${env.DEVELOPER_DIR} ` +
+                 `but no Xcode found`;
+      log.warn(mesg);
+      throw new Error(mesg);
```

Comment:
This was odd. even though `let` is block-scoped, traceur would not log me double-define `msg`